pht_federated/tests/aggregator/test_api_discovery.py

```diff
@@ -0,0 +1,184 @@
+
+
+def test_discovery_create_mixed():
+    df_titanic = pd.read_csv('./data/train_data_titanic.csv')
+    df_split = np.array_split(df_titanic, 3)
+
+    stats_df1 = statistics.get_discovery_statistics(df_split[0])
+    stats_df2 = statistics.get_discovery_statistics(df_split[1])
+    stats_df3 = statistics.get_discovery_statistics(df_split[2])
+    # print("Resulting DataSetStatistics from diabetes_dataset : {} + type {}".format(stats_df, type(stats_df)))
+
+    stats1_json = jsonable_encoder(stats_df1)
+    stats2_json = jsonable_encoder(stats_df2)
+    stats3_json = jsonable_encoder(stats_df3)
+
+    response = client.post(f"/api/proposal/{PROPOSAL_ID_MIXED}/discovery", json={
+        "item_count": stats1_json['item_count'],
+        "feature_count": stats1_json['feature_count'],
+        "column_information": stats1_json['column_information']
+    })
+    assert response.status_code == 200, response.text
+
+    response = client.post(f"/api/proposal/{PROPOSAL_ID_MIXED}/discovery", json={
+        "item_count": stats2_json['item_count'],
+        "feature_count": stats2_json['feature_count'],
+        "column_information": stats2_json['column_information']
+    })
+    assert response.status_code == 200, response.text
+
+    response = client.post(f"/api/proposal/{PROPOSAL_ID_MIXED}/discovery", json={
+        "item_count": stats3_json['item_count'],
+        "feature_count": stats3_json['feature_count'],
+        "column_information": stats3_json['column_information']
+    })
+    assert response.status_code == 200, response.text
+
+def test_discovery_get_all():
+    response = client.get(f"/api/proposal/{PROPOSAL_ID_MIXED}/discovery")
+    assert response.status_code == 200, response.text
+
+def test_discovery_get_selected():
+    response = client.get(f"/api/proposal/{PROPOSAL_ID_NUMERIC}/discovery?query={SELECTED_FEATURES}")
+    assert response.status_code == 200, response.text
+
+def test_plot_discovery():
+    response = client.get(f"/api/proposal/{PROPOSAL_ID_NUMERIC}/discovery?query={SELECTED_FEATURES}")
+    assert response.status_code == 200, response.text
+
+    discovery_summary = response.json()
+    figure_data_lst = []
+
+    data_information = discovery_summary['column_information']
+
+    for data in data_information:
+        try:
+            figure_data = {
+                "data": data['figure_data']['figure']['data'],
+                "layout": data['figure_data']['figure']['layout']
+            }
+            figure_data_lst.append(figure_data)
+        except:
+            print(f"Feature {data['title']} does not have figure_data available.")
+
+    for figure in figure_data_lst:
+        plot_figure_json(figure)
+        #print("Plotting is commented out in 'test_plot_discovery_summary_selected_features'!")
+
+def test_delete_discovery():
+    response = client.delete(f"/api/proposal/{PROPOSAL_ID_NUMERIC}/discovery")
+    assert response.status_code == 200, response.text
```

Comment:
test not only the response code but also whether the results of the aggregation etc. are correct